esp-wifi/src/lib.rs

```diff
@@ -366,7 +380,7 @@ impl EspWifiTimerSource for TimeBase {
 /// .unwrap();
 /// # }
 /// ```
-pub fn initialize(
+pub fn init(
```

Comment:
it's probably renamed to be consistent with `deinit` but not sure if we really want to one more breaking API change just for that (and all the additional noise in this PR)

Reply:
The other side of the coin is that `init` is now also consistent with esp-hal and esp-hal-embassy.

Reply:
This was my suggestion, sorry for not discussing this beforehand. It was mostly aimed at syncing with `esp_hal::init`, not so much syncing with deinit. It also has the added benefit of fewer characters to type and stops me trying to use the English spelling of initalise :D.

Whilst we probably should have done this separately, I think it's fine here provided we're all on board with the rename.